File: GoogleSearch.java
package tarea;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class GoogleSearch {

	private static WebDriver driver = null;
	private static String url = "http://www.google.com.pe";
	private static Util util = new Util();

	private static void doSearch(String searchQuery, String[] arrPages) throws InterruptedException{
		
		driver.get(url);
		
		WebElement txtSearch = driver.findElement(By.name("q"));
		
		txtSearch.sendKeys(searchQuery);
		
		util.sleep(2500);
		
		WebElement btnSearch = driver.findElement(By.xpath(".//button[@name='btnG']"));
		
		btnSearch.click();
		
		util.sleep(1500);
		
		util.printMessage("################# Search: " + searchQuery + " ###########################");
				
		for (int i = 0; i < arrPages.length; i++) {
			
			show_results_count(arrPages[i]);
			
		}
		
		util.printMessage("##############################################################################");
		
		
	}
	
	private static void show_results_count(String page) throws InterruptedException{
		
		util.printMessage("--------------------------- " + page.toUpperCase() + " TAB ---------------------------------------");
		driver.findElement(By.xpath(".//a[text()='"+ page +"']")).click();	
		util.sleep(3000);
		WebElement results = driver.findElement(By.id("resultStats"));
		util.printMessage("Results: " + util.getNumber(results.getText()));
		util.printMessage("------------------------------------------------------------------------------");
				
	}
	

		
	public static void main(String[] args) throws InterruptedException {
		// TODO Auto-generated method stub

	    driver  = new FirefoxDriver();
		
	    String[] arrSearches = {"puppies","rats"};
	    String[] arrPages = {"Vídeos","Noticias"};
	    
	    for (int i = 0; i < arrSearches.length; i++) {
			
	    	doSearch(arrSearches[i],arrPages);
	    	
		}
		
		driver.close();

	}

}

File: C-Sharp/GoogleSearch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using OpenQA.Selenium;
using OpenQA.Selenium.Firefox;
using OpenQA.Selenium.Support.UI;

namespace tarea.test
{
    class GoogleSearch
    {

        private static IWebDriver driver = null;
        private static String url = "http://www.google.com.pe";
        private static Util util = new Util();

        private static void doSearch(String searchQuery, String[] arrPages)
        {

            driver.Navigate().GoToUrl(url);

            IWebElement txtSearch = driver.FindElement(By.Name("q"));

            txtSearch.SendKeys(searchQuery);

            util.sleep(2500);

            IWebElement btnSearch = driver.FindElement(By.Name("btnG"));

            btnSearch.Click();

            util.sleep(1500);

            util.printMessage("################# Search: " + searchQuery + " ###########################");

            for (int i = 0; i < arrPages.Length; i++)
            {

                show_results_count(arrPages[i]);

            }

            util.printMessage("##############################################################################");


            }


        private static void show_results_count(String page)
        {

            util.printMessage("--------------------------- " + page.ToUpper() + " TAB ---------------------------------------");
            driver.FindElement(By.XPath(".//a[text()='"+ page +"']")).Click();
            util.sleep(3000);
            IWebElement results = driver.FindElement(By.Id("resultStats"));
            util.printMessage("Results: " + util.getNumber(results.Text));
            util.printMessage("------------------------------------------------------------------------------");

        }
        
        static void Main(String[] args)
        {
            // TODO Auto-generated method stub

            driver = new FirefoxDriver();

            String[] arrSearches = { "puppies", "rats" };
            String[] arrPages = { "Vídeos", "Noticias" };
	    
	         for (int i = 0; i<arrSearches.Length; i++) {

                   doSearch(arrSearches[i], arrPages);

             }

            driver.Close();

	    }

    }
    
}
